clock.go

```diff
@@ -89,95 +89,62 @@ func NewMock() *Mock {
 }
 
 // Add moves the current time of the mock clock forward by the specified duration.
-// This should only be called from a single goroutine at a time.
 func (m *Mock) Add(d time.Duration) {
-	// Calculate the final current time.
 	m.mu.Lock()
-	t := m.now.Add(d)
-	m.mu.Unlock()
-
-	// Continue to execute timers until there are no more before the new time.
-	for {
-		if !m.runNextTimer(t) {
-			break
-		}
-	}
-
-	// Ensure that we end with the new time.
-	m.mu.Lock()
-	m.now = t
-	m.mu.Unlock()
-
-	// Give a small buffer to make sure that other goroutines get handled.
-	gosched()
-}
-
-// Set sets the current time of the mock clock to a specific one.
-// This should only be called from a single goroutine at a time.
-func (m *Mock) Set(t time.Time) {
-	// Continue to execute timers until there are no more before the new time.
-	for {
-		if !m.runNextTimer(t) {
-			break
-		}
-	}
-
-	// Ensure that we end with the new time.
-	m.mu.Lock()
-	m.now = t
-	m.mu.Unlock()
-
-	// Give a small buffer to make sure that other goroutines get handled.
-	gosched()
+	defer m.mu.Unlock()
+	m.set(m.now.Add(d))
 }
 
-// WaitForAllTimers sets the clock until all timers are expired
+// WaitForAllTimers keeps advancing the clock until no more timers are scheduled.
+//
+// WARNING: Tickers will cause this function to continue forever until something cancels the ticker.
+// It will sleep after each bit of progress to the tickers a chance to run, but using this with
+// tickers is not recommended.
 func (m *Mock) WaitForAllTimers() time.Time {
```

Comment:
I can add a `RunNextTimer` later, that should be fairly trivial.

Reply:
RunNextTimer or a way to cancel `WaitForAllTimers` would be nice.